### DISCUSSION TITLE
Ensure unreachable

### DISCUSSION THREAD

**Discussion by axkibe** [eslint/eslint#16917] [Q&A] [answered]
Hi, is there a nifty command to say, I want this to be unreachable? 

Suppose I have a function in princple like this (but with more complicated rules)

```
function ( a )
{
  switch( a )
  {
     case 1: return 'one';
     case 2: return 'two;
     default: return 'other';
  }
};
```

And this is all nice, but I want to put an extra guard against it, it doesn't return undefined, because I forgot a return in one case. Just imagine this is a more complicated setting of if's and switches and all that.

So can I put something on the end to say, "hey eslint, I want you to complain if this is not unreachable".

Right now I do:
```
    /* eslint-disable no-unreachable */
    throw new Error( );
    /* eslint-enable *
```

So at least in runtime I get an error thrown in case something goes amiss before.

**Comment by mdjermanovic** [comment, marked as answer]
> And this is all nice, but I want to put an extra guard against it, it doesn't return undefined, because I forgot a return in one case.

Did you try enabling the [consistent-return](https://eslint.org/docs/latest/rules/consistent-return) rule? I think it would catch the errors you described. In particular, it reports reachable ends of functions (in which case an implicit `undefined` is returned) for functions that return non-undefined values in some code paths. Here's a [Playground Demo](https://eslint.org/play/#eyJ0ZXh0IjoiZnVuY3Rpb24gZm9vKCBhIClcbntcbiAgc3dpdGNoKCBhIClcbiAge1xuICAgICBjYXNlIDE6IHJldHVybiAnb25lJztcbiAgICAgY2FzZSAyOiByZXR1cm4gJ3R3byc7XG4gICAgIGNhc2UgMzogZG9Tb21ldGhpbmcoKTsgYnJlYWs7IC8vIG5vIHJldHVyblxuICAgICBkZWZhdWx0OiByZXR1cm4gJ290aGVyJztcbiAgfVxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnNpc3RlbnQtcmV0dXJuIjpbImVycm9yIl19LCJlbnYiOnsiZXM2Ijp0cnVlfX19).

**Reply by axkibe** [reply]
Thanks that seems do it quite well (it would still not cover cases where you'd "return;" in all branches but ensure it doesn't get through to the end, but that ain't the case for me)

**Comment by mdjermanovic** [comment]
> So can I put something on the end to say, "hey eslint, I want you to complain if this is not unreachable".
> 
> Right now I do:
> 
> ```
>     /* eslint-disable no-unreachable */
>     throw new Error( );
>     /* eslint-enable *
> ```
> 
> So at least in runtime I get an error thrown in case something goes amiss before.

You can also use [`"reportUnusedDisableDirectives": true`](https://eslint.org/docs/latest/use/configure/rules#report-unused-eslint-disable-comments) config option or [`--report-unused-disable-directives`](https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives) CLI option in combination with this solution, and ESLint will report a warning/error for unused `eslint-disable` directive if this code is not unreachable.

**Reply by axkibe** [reply]
Thats backward tough isn't it, since I want it to be unreachable.. and in case of unreachable have no error, but an error if it is reachable.

Also it should be documenting at the same time, like this is unreachable on purpose..
